src/slide-potentiometer-element.ts

```diff
@@ -0,0 +1,247 @@
+
+  connectedCallback() {
+    super.connectedCallback();
+    window.addEventListener('mouseup', this.up);
+    window.addEventListener('mousemove', this.mouseMove);
+    window.addEventListener('mouseleave', this.up);
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    window.removeEventListener('mouseup', this.up);
+    window.removeEventListener('mousemove', this.mouseMove);
+    window.removeEventListener('mouseleave', this.up);
+  }
+
+  private focusInput() {
+    const inputEl: HTMLInputElement | null | undefined = this.shadowRoot?.querySelector(
+      '.hide-input'
+    );
+    inputEl?.focus();
+  }
+
+  private down(): void {
+    if (!this.isPressed) {
+      this.updateCaseDisplayProperties();
+    }
+    this.isPressed = true;
+  }
+
+  private up = () => {
+    if (this.isPressed) {
+      this.isPressed = false;
+    }
+  };
+
+  private updateCaseDisplayProperties(): void {
+    const element = this.shadowRoot?.querySelector<SVGRectElement>('#sliderCase');
+    if (element) {
+      this.pageToLocalTransformationMatrix = element.getScreenCTM()?.inverse() || null;
+    }
+
+    // Handle zooming in the storybook
+    const zoom = getComputedStyle(window.document.body)?.zoom;
+    if (zoom !== undefined) {
+      this.zoom = Number(zoom);
+    } else {
+      this.zoom = 1;
+    }
+  }
+
+  private onInputValueChange(event: KeyboardEvent): void {
+    const target = event.target as HTMLInputElement;
+    if (target.value) {
+      this.updateValue(Number(target.value));
+    }
+  }
+
+  private mouseMove = (event: MouseEvent) => {
+    if (this.isPressed) {
+      this.updateValueFromXCoordinate(new DOMPointReadOnly(event.pageX, event.pageY));
+    }
+  };
+
+  private touchMove(event: TouchEvent): void {
+    if (this.isPressed) {
+      if (event.targetTouches.length > 0) {
+        const touchTarget = event.targetTouches[0];
+        this.updateValueFromXCoordinate(new DOMPointReadOnly(touchTarget.pageX, touchTarget.pageY));
+      }
+    }
+  }
+
+  private updateValueFromXCoordinate(position: DOMPointReadOnly): void {
+    if (this.pageToLocalTransformationMatrix) {
+      // Handle zoom first, the transformation matrix does not take that into account
+      let localPosition = new DOMPointReadOnly(position.x / this.zoom, position.y / this.zoom);
+      // Converts the point from the page coordinate space to the #caseRect coordinate space
+      // It also translates the units from pixels to millimeters!
+      localPosition = localPosition.matrixTransform(this.pageToLocalTransformationMatrix);
+      const caseBorderWidth = 7.5;
+      const tipPositionXinMM = localPosition.x - caseBorderWidth;
+      const mmPerIncrement = 30 / (this.maxValue - this.minValue);
+      this.updateValue(Math.round(tipPositionXinMM / mmPerIncrement));
+    }
+  }
+
+  private updateValue(value: number) {
+    let clampedValue = Math.min(value, this.maxValue);
+    clampedValue = Math.max(clampedValue, this.minValue);
```

Comment:
Optional, but if you feel like, you can move the min/max code to `utils/clamp.ts`, as it is also used by the standard potentiometer.

Reply:
Decided to create the clamp function. Applied it in the both potentiometer elements.

Reply:
Thank you!